lib/text/vtt_text_parser.js

```diff
@@ -259,17 +386,25 @@ shaka.text.VttTextParser = class {
 
   /**
    * @param {!Node} element
+   * @param {!shaka.text.Cue} rootCue
    * @param {Array.<!shaka.extern.Cue>} cues
+   * @param {!Map.<string, shaka.text.Cue>} styles
    * @private
    */
-  static generateCueFromElement_(element, rootCue, cues) {
-    const nestedCue = rootCue.clone();
+  static generateCueFromElement_(element, rootCue, cues, styles) {
+    let nestedCue = rootCue.clone();
     if (element.nodeType === Node.ELEMENT_NODE && element.nodeName) {
       const bold = shaka.text.Cue.fontWeight.BOLD;
       const italic = shaka.text.Cue.fontStyle.ITALIC;
       const underline = shaka.text.Cue.textDecoration.UNDERLINE;
       const tags = element.nodeName.split(/[ .]+/);
       for (const tag of tags) {
+        if (styles.has(tag)) {
+          nestedCue = styles.get(tag).clone();
```

Comment:
Doesn't this erase the `rootCue` that was cloned above?  Does `rootCue` not contain any styles/payload/etc?

Reply:
The payload is not necessary because it is overwritten a few lines below.
Times are copied, so it's not a problem either.
Styles will be lost when nested. Example: <b>test<i>1</i></b>
In case there is a style for i, the root style would be lost. Given the complexity to solve this, and that none of my providers use this case, I have preferred to leave it at that.